### PR TITLE
Move object safety suggestions to the end of the error

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

File: src/test/ui/issues/issue-18959.stderr [hunk 1]
```diff
@@ -4,14 +4,14 @@ error[E0038]: the trait `Bar` cannot be made into an object
...
    |
-   = help: consider moving `foo` to another trait
 note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visit <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> $DIR/issue-18959.rs:1:20
    |
 LL | pub trait Foo { fn foo<T>(&self, ext_thing: &T); }
    |                    ^^^ ...because method `foo` has generic type parameters
 LL | pub trait Bar: Foo { }
    |           --- this trait cannot be made into an object...
+   = help: consider moving `foo` to another trait
```
Comment (estebank) [inline src/test/ui/issues/issue-18959.stderr:14]: We need to account for cases where the problem is caused by a super trait and make this diagnostic better targeted.

**Comment by Manishearth** [comment]
@bors r-

I think this is causing a bunch of failures in https://github.com/rust-lang/rust/pull/88940#issuecomment-919457644

**Comment by estebank** [comment]
@bors r=petrochenkov

We now sort the suggestions before emition for consistent output.

**Pull request merged**